### PR TITLE
feat: default explore topic matches user interest

### LINKED ISSUE #929: Feature: Default explore topic should match with the first interest selected.
### Type of feature

🍕 Feature

### Current behavior

When a user chooses interests, they should be default to their interests when selecting the explore tab while logged in.

Screen Shot 2023-03-01 at 1 42 27 AM

### Suggested solution

If the interests in the DB is [python, ml], Python should be the default
If the interests in the DB is [rust, ml, Python], Rust should be the default

### Additional context

https://github.com/open-sauced/insights/issues/927

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

### Contributing Docs

- [ ] I agree to follow this project's Contribution Docs

### PR DESCRIPTION
**Pull request by shamimbinnur**
## What type of PR is this? (check all applicable)

- [ ] 🍕 Feature
- [ ] 🐛 Bug Fix
- [ ] 📝 Documentation Update
- [ ] 🎨 Style
- [ ] 🧑‍💻 Code Refactor
- [ ] 🔥 Performance Improvements
- [ ] ✅ Test
- [ ] 🤖 Build
- [ ] 🔁 CI
- [ ] 📦 Chore (Release)
- [ ] ⏩ Revert

## Description
I fetched the value of the user's interests from the database using `useFetchUser()`, which indicates the language they are interested in and set it as their default topic.

## Related Tickets & Documents
Fixes #929 

## Added tests?

- [ ] 👍 yes
- [ ] 🙅 no, because they aren't needed
- [ ] 🙋 no, because I need help

## Added to documentation?

- [ ] 📜 README.md
- [ ] 📓 docs.opensauced.pizza
- [ ] 🍕 dev.to/opensauced
- [ ] 📕 storybook
- [ ] 🙅 no documentation needed

### CODE REVIEW

File: components/organisms/TopNav/top-nav.tsx [hunk 1]
```diff
@@ -7,11 +7,15 @@ import HeaderLogo from "components/molecules/HeaderLogo/header-logo";
...
 import useSession from "lib/hooks/useSession";
 import useSupabaseAuth from "lib/hooks/useSupabaseAuth";
+import { useFetchUser } from "lib/hooks/useFetchUser";
 
 const TopNav: React.FC = () => {
   const { user } = useSupabaseAuth();
   const { onboarded } = useSession();
 
+  const { data: gitHubUser } = useFetchUser(user?.user_metadata.user_name);
+  const userInterests = gitHubUser?.interests || "javascript";
```
Comment (TAKANOME-DEV) [inline components/organisms/TopNav/top-nav.tsx:17]:
```suggestion
  const userInterests = gitHubUser?.interests?.[0] || "javascript";
```

**Review by TAKANOME-DEV — COMMENTED** [review]
@shamimbinnur if the user has multiple interests, maybe we should pick the first one. Check the screenshot below

Screenshot from 2023-03-01 15-06-50

**Comment by bdougie** [comment]
It tries to add all topics. Should just grab the first.

Screen Shot 2023-03-01 at 7 07 29 AM

File: components/organisms/TopNav/top-nav.tsx [hunk 2]
```diff
@@ -21,7 +25,7 @@ const TopNav: React.FC = () => {
             <Link className="text-sm text-light-slate-10" href={"/hub/insights"}>
             Insights Hub
             </Link>
-            <Link className="text-sm text-light-slate-10" href={"/javascript/dashboard/filter/recent"}>
+            <Link className="text-sm text-light-slate-10" href={"/"+userInterest+"/dashboard/filter/recent"}>
```
Comment (brandonroberts) [inline components/organisms/TopNav/top-nav.tsx:28]: Let's use a template literal here instead

```suggestion
            <Link className="text-sm text-light-slate-10" href={`${userInterest}/dashboard/filter/recent`}>
```

**Review by brandonroberts — CHANGES_REQUESTED** [review]

**Review by TAKANOME-DEV — APPROVED** [review]
LGTM :+1:

**Review by bdougie — APPROVED** [review]
Looks good

**Review by brandonroberts — APPROVED** [review]
LGTM. We'll need to hook this into the user settings being updated also, but separate from this

**Comment by shamimbinnur** [comment]
@brandonroberts  I just noticed this [line](https://github.com/open-sauced/insights/blob/fddd634467aa16da18c6111aada1a8ed42ef248a/components/organisms/TopNav/top-nav.tsx#L30) has not been updated. Was it intentional?

**Comment by brandonroberts** [comment]
@shamimbinnur it was not intentional. It got changed accidentally while resolving some merge conflicts. It should be reapplied

**Pull request merged**